### PR TITLE
core: Remove some unnecessary code added in #15627

### PR DESCRIPTION
**Pull request by Flawake**

### CODE REVIEW

File: core/src/player.rs [hunk 1]
```diff
@@ -735,20 +735,14 @@ impl Player {
...
-                    context,
-                    *context.mouse_position,
-                    HitTestOptions::MOUSE_PICK,
-                ) {
-                    if let Some(Value::Object(obj)) = Some(disp_obj).map(|obj| obj.object()) {
-                        let mut activation = Activation::from_stub(
-                            context.reborrow(),
-                            ActivationIdentifier::root("[ContextMenu]"),
-                        );
+                if let Some(Value::Object(obj)) = Some(disp_obj).map(|obj| obj.object()) {
```
Comment (Lord-McSweeney) [inline core/src/player.rs:738]: Can this line just be
```
                if let Some(Value::Object(obj)) = disp_obj.object() {
```
?

Comment (Flawake) [reply core/src/player.rs]: Fixed that, also refactored the rest of the function.

**Review by adrian17 — APPROVED** [review]

**Comment by Flawake** [comment]
> @Flawake do you want to resolve the conflict?


Currently somewhere between paris and the netherlands. But I will do it in the coming days.

**Comment by torokati44** [comment]
I did a rebase for you.
Here's the range-diff:
image

The only meaningful change was the removal of `.reborrow()` after `context`, and of the first lifetime on `UpdateContext`. Plus some formatting.

**Pull request merged**